spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/instrument/web/client/TraceWebClientBeanPostProcessor.java

```diff
@@ -309,10 +288,57 @@ public Context currentContext() {
 			return this.context;
 		}
 
-		void handleReceive(@Nullable ClientResponse res, @Nullable Throwable error) {
-			ClientResponseWrapper response = res != null ? new ClientResponseWrapper(res)
-					: null;
-			this.handler.handleReceive(response, error, clientSpan);
+	}
+
+	static class TraceWebClientSubscription extends AtomicBoolean
+			implements Subscription {
+
+		static final Exception CANCELLED_ERROR = new CancellationException("CANCELLED") {
+			@Override
+			public Throwable fillInStackTrace() {
+				return this; // stack trace doesn't add value here
+			}
+		};
+
+		final AtomicReference<Span> pendingSpan;
+
+		final Subscription delegate;
+
+		TraceWebClientSubscription(Subscription delegate,
+				AtomicReference<Span> pendingSpan) {
+			this.delegate = delegate;
+			this.pendingSpan = pendingSpan;
+		}
+
+		@Override
+		public void request(long n) {
+			if (compareAndSet(false, true)) {
```

Comment:
@adriancole looks strange, you apply only first request. Why to skip others?

Reply:
because only concern was if this signal happened.. if distracting, I can do it on the others.

Reply:
actually I need to do it here, as the most important thing is if it happened. doing elsewhere would invalidate that. probably if this is confusing I can put a comment.

Reply:
`compareAndSet(false, true)` part is ok - this value is used in `cancel()`, but I do not understand `if ()` part, why it would be invalidated?

Reply:
I mean this `Subscriber` should not impact the way signals are propagated but only trigger appropriate `brave`  API

Reply:
good idea. I will switch this in next change

Reply:
https://github.com/spring-cloud/spring-cloud-sleuth/pull/1581